### DISCUSSION TITLE
共享农场收益算法优化方案（待讨论）

### DISCUSSION THREAD

**Discussion by welllom** [famland/website#137] [Ideas]
# 共享农场收益算法优化方案（待讨论）

共享农场每个周期的收益率采取固定APY收益率。在同一个周期内，Famland的APY收益率不会因为农场内某个Famland提取收益而改变。

单个Famland的收益，在每个周期结算完成后，Famalnd待领取的收益将直接显示在界面上，玩家在一个周期内任何时候领取，均可以按照显示的固定YL数额，领取收益。这个周期结束后，玩家若没有领取收益，未领取的收益将当前周期结束后重新结算分配。每个周期每个famland只能领取一次收益。

每个周期结算时，系统根据农场上一个周期农场剩余分发池额度，一次性累加根据流速计算增加的分发池额度，作为下个周期分发池起始的基数。

单个Famland的收益的收益额度：

单个Famland周期的收益额度 = （Famland做市本金加成价值 /  农场所有Famland总的做市本金加成价值）*  农场上一个周期结算时间的分发池基数 * 30%

农场APY计算规则：

农场年化APY=（农场上一个周期结算时间的分发池基数*30% * 上个周期YL基准价 / 农场所有Famland总的做市本金加成价值 )  * 365  ，封顶值为3000%。

注：Famland做市本金加成价值 = （基本田做市本金实际价值+原始田做市本金实际价值 ）* 肥力加成系数  + 野田做市本金实际价值

**Comment by yancanlou** [comment]
这个收益算法优化方案，其APY比较稳定与实际收益接近，但其放弃了很多白皮书关于收益大小的表述理念，明显的就是收益与时长无关，与农场流速正相关，即与共享值基本正相关，且必须每日签到。
而原来收益法，即使共享值高若不会及时抢点是没有什么大收益，会被场内其它玩家抢走，空留时长。
若是要改，到时改前要通知到位。因为：
这个一改，原生农场影响会很大，而共享农场影响会小点，单人农场无场内博弈几无影响。
改后最大的变化是收益值与上次收取积累时长无关，那些都没收取的玩家要在改前收取一次，否则将浪费了时长积累。
而改后以后每天也要任意时间收取一次，否则将收弃收益回收分发池给第二天再分配。

**Comment by yancanlou** [comment]
这个提的新的收益算法现在不是紧迫要改的，改了下，很多白皮书概念都要改，可以先搁置起来[吃瓜]，不急的事，没钱时先不动。
还存在定点结算引起的生态问题等，更没必要去花心思整它了，浪费时间浪费金钱。

**Comment by yancanlou** [comment]
原则野田没有做市时长是不能参与收益的，新收益法又取消原收益法做市时长算法。
问题出现的要求显然又要针对野田这块增加做市时长，硬要改只能尴尬的组合在一起😓 
如：分发池里30%额去除野田资金参与新收益法结算，另拿出分发池3%采用现收益算法结算！
浪费开发钱不？

**Comment by famland16816** [comment]
建议
野田资金不参与，单田收益计算。
因为野田做市，没有，贡献增发量。
只是贡献了，共享金额度。

这个增发量是按基本田来的。那么分配按基本田来分配才是最公平的。
一块基本田相当于一个人。这就是以人为本。

野田没有贡献增发量。却比贡献增发量的基本田多收十倍。（满额做市10000U的情况下）
那么对其他没有做市野田的基本田是不公平的。
打个比方。
我一万U，我在本周期结算前存入野田。
结算后，我再去，取回。

那么这次结算收益，我是不是比其他没有做市野田的收益高。抢了其他农田的，分配额度。因为按照这个方案的算法。单田收益是按资本占比，分配的。

按照常理我野田投入了就应该有收入。
其实这个野田不参与这个单田收益分配。也不是说，完全，没有带来收入。
因为野田投入，可以，提升共享金额度。
而共享金额度可以用来，提升共享值。
那么共享直提高了，可以，提高农场的流速，从而提高分发池分配量。也就提高了单田的收益。

我们也可以这样理解。
这个投入野田一万U的，是先富的人。他不缺钱，也用不完。那么他可以拿来做贡献。
这个做贡献，还不会让他的钱减少。而且他帮大家提升共享值，这个大家和他的收入都会提高的。
这样可以理解可以接受吧。

**关于这个共享金额度。**

说一个大胆可能还不太成熟的想法
我们这个总的理念设计是充满哲学大智慧的。

这个野田每投入1U就可以给其名下任意一块儿田来提升共享值。这是这个理念的创新。按照常规，你，在哪块田投入那么只能给哪一块提升共享值。
那么可不可以再大胆一点？

既然叫共享金额度。那么就不只是你自己可以享受这个额度。
而是你所在的整个农场的农田都可以享受这个共享金额度。
这个就是有那个，先富起来的人带动帮助后富的人的意思。
先富带后富，不能是，把先富的钱分给后富的人。

这个野田投的钱还是属于先富人的。
但这个共享金额度，可以分给大家，共享。
帮助大家，提升共享值。从而帮农场获得更高的流速，获得更多的分配量。这个分配量又是大家共享的。
完美的体现，人人为我，我为人人。
